components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -147,6 +150,9 @@ message WorkspaceContext {
     message Git {
         string normalized_context_url = 1;
         string commit = 2;
+
+        // git_provider is the git provider, e.g. 'github', 'gitlab', self hosted, etc
+        string git_provider = 3;
```

Comment:
Only if it's really easy to add, otherwise I'll just discard it
This is what I;m doing currently client side
```ts
    const toWorkspaceData = (ws: Workspace) => {
        const url = new URL(ws.context!.contextUrl);
        const provider = url.host.replace(/\..+?$/, ''); // remove '.com', etc
        const matches = url.pathname.match(/[^/]+/g)!; // match /owner/repo
        const owner = matches[0];
        const repo = matches[1];
        return {
            provider,
            owner,
            repo,
            id: ws.workspaceId,
            contextUrl: ws.context!.contextUrl,
            workspaceUrl: ws.status!.instance!.status!.url,
            phase: ws.status?.instance?.status?.phase
        };
    };
```

Reply:
It may also be useful to specify expected naming here. For example, to have it as `github.io`, `gitlab.com` etc. That would basically mean that it's nearly always a domain name. This would provide at least some expectation for the consumer on the range of values to expect here

Reply:
yep, this is most relevant for self hosted I think then maybe it would better to make it an object with `{host:string, type:string}` sounds good?